docs/proposals/api-grouping.md

```diff
@@ -0,0 +1,375 @@
+/apis/componentconfig/v1alpha1/
+```
+
+Those api groups have nearly 50 "top-level" types: types that are
+visible to users and which you can GET and/or POST by themselves
+(not nested in some other type).
+
+```
+/api/v1
+	persistentVolume
+	persistentVolumeClaim
+	pod
+	podTemplate
+	replicationController
+	service
+	serviceAccount
+	endpoints
+	node
+	namespace
+	binding
+	event
+	limitRange
+	resourceQuota
+	secret
+	componentStatus
+	configMap
+
+/apis/extensions/v1beta1
+	scale
+	horizontalPodAutoscaler
+	thirdPartyResource
+	thirdPartyResourceData
+	deployment
+	daemonSet
+	job
+	ingress
+	clusterAutoscaler
+	replicaSet
+	scaleStatus
+	scale
+
+/apis/abac/v0
+	policy // not currently available through API, but might be some day.
+
+/apis/abac/v1beta1
+	policy // not currently available through API, but might be some day.
+
+/apis/authorization/v1beta1
+	subjectAccessReview
+	selfSubjectAccessReview
+	localSubjectAccessReview
+
+/apis/componentconfig/v1alpha1/
+	kubeProxyConfiguration
+
+/apis/metrics/v1alpha1
+	rawNode
+	rawPod
+```
+
+## Upcoming Additions
+
+New top level objects being considered:
+- `template` and `parameters` are added by #18215.  Other things do not
+   depend directly on them.  They work as an independent template expansion
+   service.
+- `Node`, `Pod` metrics types are being added, tenatively to `metrics` group. (see #18824)
+- `PetSet`: https://github.com/kubernetes/kubernetes/pull/18016 
+- `heartbeat` by #14735
+- `externalServiceProvider` by #13216
+- `podSecurityPolicy` and maybe `PodSecurityPolicyBinding` by #7893 
+- `workflow` by  #18827
+- `scheduledJob` by #11980
+- `metadataPolicy` by #18262
+- `ingress` may change, and probably will move out of the main api group.
+- `podStatusResult` should maybe move to a kubelet API group?
+
+## Inter-Kind references
+
+I list some dependencies here.
+
+- `ScheduledJob` depends on `Workflow` and `Job`; `Workflow` depends on `Job`
+- `Job`, `ReplicationController`, `ReplicaSet`, `Deployment` and `DaemonSet` make `Pods` via PodTemplates.
+  - this is a strong dependency.
+- Pods that use PersistentVolumeClaimVolumeSource depend on PersistentVolumeClaim object (`ClaimName`).
+- PersistentVolumeClaims depend on PersistentVolume names.
+- PersistentVolumes depend on PersistentVolumeClaims via ClaimRef.
+- `GlusterfsVolumeSource` of a Pod depends on `Endpoints`.
+- Pods depend on `Secrets` several ways: `SecretVolumeSource`, EnvVars.
+- `Pods` depend on `ServiceAccounts` via `ServiceAccountName`.
+  - this seems like a pretty strong dependency.
+- `Pods` depend on `Nodes` in that `Pods` have a `NodeName`.
+- `Pods` with an `RBDVolumeSource` or `CephFSVolumeSource` or `FlexVolumeSource` depend on `Secrets` to store a secret.
+- `Pods` depend on `Secrets` for `imagePullSecrets`.
+- `ServiceAccounts` refer to `Secrets`.
+- `Pods` can depend on `ConfigMap` or `Secret` via env vars, etc.
+- `ReplicationControllers` depend on `Pods` via a `PodTemplate`.
```

Comment:
HorizontalPodAutoscaler depends on the "Scale" subresource on objects in multiple groups - ReplicationController, Deployment, or PetSet